mantidproject/mantid
# Load Widget for new Elemental Analysis GUI

**Description of work.**

A load widget for the new Elemental Analysis GUI. Users should be able to add a single run, a range of runs or co-add a series of runs to work on. 

**To test:**

1. Code review

2. Before starting Mantid go to https://github.com/mantidproject/mantid/blob/master/Framework/Properties/Mantid.properties.template#L24 and add Muon/Elemental_Analysis_2.py to the end of the line
3. Start Mantid
4. Open the elemental analysis 2 GUI - it should appear as per screenshot below
![Screenshot (50)](https://user-images.githubusercontent.com/55837273/99068632-c4b51c80-25a4-11eb-9d33-4b74ad974d57.png)

5. Download & unzip the 12 dat files for runs 9997-9 (held in attached .zip folder) and add to Manage User Directories
[29380 tests.zip](https://github.com/mantidproject/mantid/files/5536599/29380.tests.zip)

6. Enter 9997 in the Loading Box and press return. Should load a group workspace named 9997 containing 4 workspaces, one for each detector for that run.
7. Click Clear All and the Loading Box should clear 
8. Enter 9998-9 in the Loading box and press co-add. Should add group workspaces 9998, 9999 each containing 4 workspaces. A third groupworkspace named 9998-9999 should also appear and be an amalgamation of the data from 9998 and 9999.
9. Clear all the workspaces.
10. Enter 9997 in the loading box and press return. Then click the increment button (>). Run 9998 should load.
11. Clear all workspaces. Click the decrement button (<). Run 9997 should load. 


Fixes #29380 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
